2_run/src/run_glm3.R

```diff
@@ -0,0 +1,150 @@
+#' @title Munge meteo and add burn-in and burn-out
+#' @description function to read in the raw meteorological data
+#' form the file specified in the meteo_xwalk, filter it based
+#' on the specified time_period, and add burn-in and burn-out
+#' periods by mirroring the data. From Jordan's GLM projection code: 
+#' https://github.com/jread-usgs/lake-temperature-process-models/blob/master/3_run/src/run_glm_utils.R#L62-L76
+#' @param meteo_xwalk a lake-gcm-time_period crosswalk table with the 
+#'  meteo file name and hash for each model run
+#'  @param begin the begin date for the model time_period
+#'  @param end the end date for the model_time_period
+#'  @param burn_in length of burn-in period, in days
+#'  @param burn_out length of burn-out period, in days
+#'  @return a munged dataframe of meteorological data that includes
+#'  burn-in and burn-out time
+munge_meteo_w_burn_in_out <- function(meteo_xwalk, begin, end, burn_in, burn_out){
+  meteo_data <- arrow::read_feather(meteo_xwalk$meteo_fl) %>%
+    mutate(time = as.Date(time)) %>% filter(time >= as.Date(begin) & time <= as.Date(end)) %>%
+    mutate(Rain = case_when(Snow > 0 ~ 0, TRUE ~ Rain))
+  # create a burn-in. We're going to mirror the data of the first year:
+  burn_in_data <- meteo_data[2:burn_in, ] %>% arrange(desc(time)) %>%
+    mutate(time = seq(from = meteo_data[1,]$time-burn_in+1, by = 'day', length.out = length(2:burn_in)))
+  # bind with real data, prior to the begin date. We're not duplicating dates, hence the "2"
+  meteo_data <- bind_rows(burn_in_data, meteo_data)
+  # create a burn-out. Mirror the final data:
+  burn_out_data <- tail(meteo_data, burn_out) %>% head(-1L) %>% arrange(desc(time)) %>%
+    mutate(time = seq(from = tail(meteo_data,1)$time+1, by = 'day', length.out = length(2:burn_out)))
+  # bind with real data, at the end of the sequence
+  meteo_data <- bind_rows(meteo_data, burn_out_data)
+  return(meteo_data)
+}
+
+#' @title function to extract and write glm output following a model run
+#' @description Extract temperature and ice data from the model
+#' run and save to the export feather file. 
+#' From Jordan's GLM projection code: 
+#' https://github.com/jread-usgs/lake-temperature-process-models/blob/master/3_run/src/run_glm_utils.R#L89-L101
+#' @param sim_lake_dir the directory for the model simulation
+#' @param nml_obj the complete nml object for the model run
+#' @param export_fl the filepath for the exported feather file
+extract_glm_output <- function(sim_lake_dir, nml_obj, export_fl) {
+  out_dir <- glmtools::get_nml_value(nml_obj, arg_name = 'out_dir')
+  out_fn <- paste0(glmtools::get_nml_value(nml_obj, 'out_fn'), '.nc')
+  nc_filepath = file.path(sim_lake_dir, out_dir, out_fn)
```

Comment:
tiny, but update 
`nc_filepath <- file.path(sim_lake_dir, out_dir, out_fn)`
for consistency in assignment. I think there are a mix of some of these "=" in the older repos that come from copy-pasting past function snippets. But a good opportunity here to clean up